nsxt/resource_nsxt_policy_security_policy_rule.go

```diff
@@ -0,0 +1,252 @@
+		return handleReadError(d, "SecurityPolicyRule", fmt.Sprintf("%s/%s", policyPath, id), err)
+	}
+
+	securityPolicyRuleModelToSchema(d, rule)
+	return nil
+}
+
+func securityPolicyRuleModelToSchema(d *schema.ResourceData, rule model.Rule) {
+	d.Set("display_name", rule.DisplayName)
+	d.Set("description", rule.Description)
+	d.Set("path", rule.Path)
+	d.Set("notes", rule.Notes)
+	d.Set("logged", rule.Logged)
+	d.Set("log_label", rule.Tag)
+	d.Set("action", rule.Action)
+	d.Set("destinations_excluded", rule.DestinationsExcluded)
+	d.Set("sources_excluded", rule.SourcesExcluded)
+	if rule.IpProtocol == nil {
+		d.Set("ip_version", "NONE")
+	} else {
+		d.Set("ip_version", rule.IpProtocol)
+	}
+	d.Set("direction", rule.Direction)
+	d.Set("disabled", rule.Disabled)
+	d.Set("revision", rule.Revision)
+	setPathListInSchema(d, "source_groups", rule.SourceGroups)
+	setPathListInSchema(d, "destination_groups", rule.DestinationGroups)
+	setPathListInSchema(d, "profiles", rule.Profiles)
+	setPathListInSchema(d, "services", rule.Services)
+	setPathListInSchema(d, "scope", rule.Scope)
+	d.Set("sequence_number", rule.SequenceNumber)
+	d.Set("nsx_id", rule.Id)
+	d.Set("rule_id", rule.RuleId)
+
+	setPolicyTagsInSchema(d, rule.Tags)
+}
+
+func resourceNsxtPolicySecurityPolicyRuleUpdate(d *schema.ResourceData, m interface{}) error {
+	connector := getPolicyConnector(m)
+
+	id := d.Id()
+	if id == "" {
+		return fmt.Errorf("Error obtaining Security Policy Rule ID")
+	}
+
+	policyPath := d.Get("policy_path").(string)
+	log.Printf("[INFO] Updating Security Policy Rule with ID %s under policy %s", id, policyPath)
+	domain := getDomainFromResourcePath(policyPath)
+	policyID := getPolicyIDFromPath(policyPath)
+
+	client := securitypolicies.NewRulesClient(getSessionContext(d, m), connector)
+	rule := securityPolicyRuleSchemaToModel(d, id)
+	err := client.Patch(domain, policyID, id, rule)
+	if err != nil {
+		return handleUpdateError("SecurityPolicyRule", fmt.Sprintf("%s/%s", policyPath, id), err)
+	}
+
+	return resourceNsxtPolicySecurityPolicyRuleRead(d, m)
+}
+
+func resourceNsxtPolicySecurityPolicyRuleDelete(d *schema.ResourceData, m interface{}) error {
+	id := d.Get("nsx_id").(string)
+	if id == "" {
+		return fmt.Errorf("Error obtaining Security Policy Rule ID")
+	}
+
+	connector := getPolicyConnector(m)
+
+	policyPath := d.Get("policy_path").(string)
+	log.Printf("[INFO] Deleting Security Policy Rule with ID %s under policy %s", id, policyPath)
+	domain := getDomainFromResourcePath(policyPath)
+	policyID := getPolicyIDFromPath(policyPath)
+
+	client := securitypolicies.NewRulesClient(getSessionContext(d, m), connector)
+	return client.Delete(domain, policyID, id)
+}
+
+func nsxtSecurityPolicyRuleImporter(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	importID := d.Id()
+	rd, err := nsxtPolicyPathResourceImporterHelper(d, m)
+	if err != nil {
+		return rd, err
+	}
+	d.Set("policy_path", importID[:strings.Index(importID, "rule")-1])
```

Comment:
This requires validation, since we can't trust the user to always provide correct path. `rule` might not be present there at all, in which case provider will crash

Reply:
Good one. Done.